sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -264,8 +262,80 @@ class SQLContext(@transient val sparkContext: SparkContext)
   }
 
   @DeveloperApi
-  def applySchema(rowRDD: JavaRDD[Row], schema: StructType): DataFrame = {
-    applySchema(rowRDD.rdd, schema);
+  def createDataFrame(rowRDD: JavaRDD[Row], schema: StructType): DataFrame = {
+    createDataFrame(rowRDD.rdd, schema)
+  }
+
+  /**
+   * Creates a [[DataFrame]] from an [[RDD]] containing [[Row]]s by applying
+   * a seq of names of columns to this RDD, the data type for each column will
+   * be inferred by the first row.
+   *
+   * @param rowRDD an RDD of Row
+   * @param columns names for each column
+   * @return DataFrame
+   */
+  def createDataFrame(rowRDD: RDD[Row], columns: Seq[String]): DataFrame = {
```

Comment:
If nested structures are not supported, you can use `typeOfObject` in `ScalaReflection` to simplify the logic at here. If we want to support nested structures, it will be better to generalize the logic used by JSON and use that at here.